src/platform_impl/linux/wayland/window.rs

```diff
@@ -154,11 +154,18 @@ impl Window {
             Some(Fullscreen::Exclusive(_)) => {
                 panic!("Wayland doesn't support exclusive fullscreen")
             }
-            Some(Fullscreen::Borderless(RootMonitorHandle {
-                inner: PlatformMonitorHandle::Wayland(ref monitor_id),
-            })) => frame.set_fullscreen(Some(&monitor_id.proxy)),
-            #[cfg(feature = "x11")]
-            Some(Fullscreen::Borderless(_)) => unreachable!(),
+            Some(Fullscreen::Borderless(monitor)) => {
+                let monitor = if let Some(RootMonitorHandle {
+                    inner: PlatformMonitorHandle::Wayland(monitor),
+                }) = monitor
+                {
+                    Some(monitor.proxy)
+                } else {
+                    None
+                };
```

Comment:
This looks suspiciously `map`y.

Reply:
The reason why there's no `map` its' because there's an `X` variant on `PlatformMonitorHandle` so I need `if let` anyway. And that `X` variant is presented based on feature, so in one case you must put `unreachable!` there, so I decided to you that, since shorter.